code/game/objects/items/devices/scanners.dm

```diff
@@ -369,11 +369,28 @@ GENE SCANNER
 		if(tdelta < (DEFIB_TIME_LIMIT * 10))
 			to_chat(user, "<span class='alert'><b>Subject died [DisplayTimeText(tdelta)] ago, defibrillation may be possible!</b></span>")
 
+	var/mob/living/carbon/human/target_human = M
+	var/perpname = target_human.get_face_name(target_human.get_id_name())
+	var/datum/data/record/med_rec = find_record("name", perpname, GLOB.data_core.medical)
+	var/list/disease_format = list()
+
 	for(var/thing in M.diseases)
 		var/datum/disease/D = thing
 		if(!(D.visibility_flags & HIDDEN_SCANNER))
+			if(speaking_mob)
+				var/obj/item/radio/headset/radio = new(speaking_mob)
+				radio.keyslot = new /obj/item/encryptionkey/headset_med
+				radio.subspace_transmission = TRUE
+				radio.canhear_range = 0
+				radio.recalculateChannels()
+				radio.talk_into(speaking_mob, "Warning! Crewmember [M] has [D.form]: [D.name] ([D.stage]/[D.max_stages]). Possible cure is [D.cure_text]", RADIO_CHANNEL_MEDICAL)  // alert
```

Comment:
I like this idea, but i worry about it spamming the channel. this should have a cooldown

Reply:
oh good idea

Reply:
may also be wise to broadcast transmission vectors!